In [38]:
import pandas as pd

--------------------------------------------------------------------------------------

### Dataset "homicidios"

In [39]:
hechos = pd.read_excel('datasets/homicidios.xlsx', sheet_name='HECHOS')
hechos.head(3) 

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA HH  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00  4   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00  1   
2  2016-0003           1 2016-01-03  2016   1   3  07:00:00  7   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE  ...  Altura  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...     NaN   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...     NaN   
2                         AV ENTRE RIOS 2034       AVENIDA  ...  2034.0   

                                 Cruce  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                  DE LOS CORRALES AV.   
2                                  NaN   

                               Dirección Normalizada COMUNA  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                ENTRE RIOS AV. 2034      1   

                                XY (CABA)         pos x         pos y  \
0   Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1   Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2  Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   

    PARTICIPANTES VICTIMA    ACUSADO  
0       MOTO-AUTO    MOTO       AUTO  
1  AUTO-PASAJEROS    AUTO  PASAJEROS  
2       MOTO-AUTO    MOTO       AUTO  

[3 rows x 21 columns]

* Tal como vimos en el EDA se eliminaran las columnas "Cruce" y "Altura"

In [40]:
hechos.drop(['Altura', 'Cruce'], axis=1 , inplace=True)
hechos.head()

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA HH  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00  4   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00  1   
2  2016-0003           1 2016-01-03  2016   1   3  07:00:00  7   
3  2016-0004           1 2016-01-10  2016   1  10  00:00:00  0   
4  2016-0005           1 2016-01-21  2016   1  21  05:20:00  5   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE             Calle  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  PIEDRA BUENA AV.   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ    PAZ, GRAL. AV.   
2                         AV ENTRE RIOS 2034       AVENIDA    ENTRE RIOS AV.   
3      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA    LARRAZABAL AV.   
4  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA      SAN JUAN AV.   

                               Dirección Normalizada  COMUNA  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...       8   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.       9   
2                                ENTRE RIOS AV. 2034       1   
3          LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.       8   
4            SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.       1   

                                 XY (CABA)         pos x         pos y  \
0    Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1    Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2   Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3    Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4  Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   

    PARTICIPANTES VICTIMA    ACUSADO  
0       MOTO-AUTO    MOTO       AUTO  
1  AUTO-PASAJEROS    AUTO  PASAJEROS  
2       MOTO-AUTO    MOTO       AUTO  
3         MOTO-SD    MOTO         SD  
4  MOTO-PASAJEROS    MOTO  PASAJEROS

In [41]:
hechos.shape

(696, 19)

In [42]:
hechos['FECHA'].head(5)

0   2016-01-01
1   2016-01-02
2   2016-01-03
3   2016-01-10
4   2016-01-21
Name: FECHA, dtype: datetime64[ns]

In [43]:
hechos.isnull().sum()

ID                       0
N_VICTIMAS               0
FECHA                    0
AAAA                     0
MM                       0
DD                       0
HORA                     0
HH                       0
LUGAR_DEL_HECHO          0
TIPO_DE_CALLE            0
Calle                    1
Dirección Normalizada    8
COMUNA                   0
XY (CABA)                0
pos x                    0
pos y                    0
PARTICIPANTES            0
VICTIMA                  0
ACUSADO                  0
dtype: int64

El dataframe cuanta con 696 filas y 19 columnas de las cuales voy a borrar las que no utilizare

In [44]:
hechos.drop(['AAAA', 'MM','DD','HH','XY (CABA)'], axis=1 , inplace=True)
hechos.head()

ID  N_VICTIMAS      FECHA      HORA  \
0  2016-0001           1 2016-01-01  04:00:00   
1  2016-0002           1 2016-01-02  01:15:00   
2  2016-0003           1 2016-01-03  07:00:00   
3  2016-0004           1 2016-01-10  00:00:00   
4  2016-0005           1 2016-01-21  05:20:00   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE             Calle  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  PIEDRA BUENA AV.   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ    PAZ, GRAL. AV.   
2                         AV ENTRE RIOS 2034       AVENIDA    ENTRE RIOS AV.   
3      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA    LARRAZABAL AV.   
4  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA      SAN JUAN AV.   

                               Dirección Normalizada  COMUNA         pos x  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...       8  -58.47533969   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.       9  -58.50877521   
2                                ENTRE RIOS AV. 2034       1  -58.39040293   
3          LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.       8  -58.46503904   
4            SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.       1  -58.38718297   

          pos y   PARTICIPANTES VICTIMA    ACUSADO  
0  -34.68757022       MOTO-AUTO    MOTO       AUTO  
1  -34.66977709  AUTO-PASAJEROS    AUTO  PASAJEROS  
2  -34.63189362       MOTO-AUTO    MOTO       AUTO  
3  -34.68092974         MOTO-SD    MOTO         SD  
4  -34.62246630  MOTO-PASAJEROS    MOTO  PASAJEROS

--------------------------------------------------------------------------

### Dataset "homicidios"

In [45]:
victimas = pd.read_excel('datasets/homicidios.xlsx', sheet_name='VICTIMAS')
victimas.head(5)

ID_hecho      FECHA  AAAA  MM  DD        ROL VICTIMA       SEXO EDAD  \
0  2016-0001 2016-01-01  2016   1   1  CONDUCTOR    MOTO  MASCULINO   19   
1  2016-0002 2016-01-02  2016   1   2  CONDUCTOR    AUTO  MASCULINO   70   
2  2016-0003 2016-01-03  2016   1   3  CONDUCTOR    MOTO  MASCULINO   30   
3  2016-0004 2016-01-10  2016   1  10  CONDUCTOR    MOTO  MASCULINO   18   
4  2016-0005 2016-01-21  2016   1  21  CONDUCTOR    MOTO  MASCULINO   29   

   FECHA_FALLECIMIENTO  
0  2016-01-01 00:00:00  
1  2016-01-02 00:00:00  
2  2016-01-03 00:00:00  
3                   SD  
4  2016-02-01 00:00:00

* Procedo a eliminar las columnas que no voy a utilizar 

In [46]:
victimas.drop(['AAAA', 'MM','DD', 'FECHA','VICTIMA'], axis=1 , inplace=True)
victimas.head()

ID_hecho        ROL       SEXO EDAD  FECHA_FALLECIMIENTO
0  2016-0001  CONDUCTOR  MASCULINO   19  2016-01-01 00:00:00
1  2016-0002  CONDUCTOR  MASCULINO   70  2016-01-02 00:00:00
2  2016-0003  CONDUCTOR  MASCULINO   30  2016-01-03 00:00:00
3  2016-0004  CONDUCTOR  MASCULINO   18                   SD
4  2016-0005  CONDUCTOR  MASCULINO   29  2016-02-01 00:00:00

In [47]:
victimas.isnull().sum()

ID_hecho               0
ROL                    0
SEXO                   0
EDAD                   0
FECHA_FALLECIMIENTO    0
dtype: int64

In [48]:
victimas.shape

(717, 5)

-------------------------------------------------------------------------------------------------------

## Generamos un merge para combinar los dos dataframes en uno solo 

In [49]:
df = pd.merge(hechos, victimas, left_on='ID', right_on='ID_hecho', how='inner')
df.set_index('ID', inplace=True)
df.head(5)

N_VICTIMAS      FECHA      HORA  \
ID                                           
2016-0001           1 2016-01-01  04:00:00   
2016-0002           1 2016-01-02  01:15:00   
2016-0003           1 2016-01-03  07:00:00   
2016-0004           1 2016-01-10  00:00:00   
2016-0005           1 2016-01-21  05:20:00   

                                     LUGAR_DEL_HECHO TIPO_DE_CALLE  \
ID                                                                   
2016-0001  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA   
2016-0002          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ   
2016-0003                         AV ENTRE RIOS 2034       AVENIDA   
2016-0004      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA   
2016-0005  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA   

                      Calle  \
ID                            
2016-0001  PIEDRA BUENA AV.   
2016-0002    PAZ, GRAL. AV.   
2016-0003    ENTRE RIOS AV.   
2016-0004    LARRAZABAL AV.   
2016-0005      SAN JUAN AV.   

                                       Dirección Normalizada  COMUNA  \
ID                                                                     
2016-0001  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...       8   
2016-0002               PAZ, GRAL. AV. y DE LOS CORRALES AV.       9   
2016-0003                                ENTRE RIOS AV. 2034       1   
2016-0004          LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.       8   
2016-0005            SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.       1   

                  pos x         pos y   PARTICIPANTES VICTIMA    ACUSADO  \
ID                                                                         
2016-0001  -58.47533969  -34.68757022       MOTO-AUTO    MOTO       AUTO   
2016-0002  -58.50877521  -34.66977709  AUTO-PASAJEROS    AUTO  PASAJEROS   
2016-0003  -58.39040293  -34.63189362       MOTO-AUTO    MOTO       AUTO   
2016-0004  -58.46503904  -34.68092974         MOTO-SD    MOTO         SD   
2016-0005  -58.38718297  -34.62246630  MOTO-PASAJEROS    MOTO  PASAJEROS   

            ID_hecho        ROL       SEXO EDAD  FECHA_FALLECIMIENTO  
ID                                                                    
2016-0001  2016-0001  CONDUCTOR  MASCULINO   19  2016-01-01 00:00:00  
2016-0002  2016-0002  CONDUCTOR  MASCULINO   70  2016-01-02 00:00:00  
2016-0003  2016-0003  CONDUCTOR  MASCULINO   30  2016-01-03 00:00:00  
2016-0004  2016-0004  CONDUCTOR  MASCULINO   18                   SD  
2016-0005  2016-0005  CONDUCTOR  MASCULINO   29  2016-02-01 00:00:00

In [50]:
df.shape

(717, 18)